pkg/controllers/state/cluster.go

```diff
@@ -73,6 +75,8 @@ func NewCluster(clk clock.Clock, client client.Client) *Cluster {
 		daemonSetPods:             sync.Map{},
 		nodeNameToProviderID:      map[string]string{},
 		nodeClaimNameToProviderID: map[string]string{},
+		lastPodEvents:             map[string]time.Time{},
+		initTime:                  clk.Now(),
```

Comment:
I wonder how much you really need this init time if you can just use time.Now() for the node creation if you haven't seen any pod events yet.